File: app/src/main/java/com/github/mrvilkaman/presentationlayer/fragments/testfrags/FragScreenComponent.java
package com.github.mrvilkaman.presentationlayer.fragments.testfrags;

import dagger.Component;
import dagger.Module;
import com.github.mrvilkaman.di.PerScreen;
import com.github.mrvilkaman.presentationlayer.activities.ActivityCoreComponent;

@PerScreen
@Component(dependencies = ActivityCoreComponent.class,
		modules = {FragScreenComponent.FragScreenModule.class})
public interface FragScreenComponent {
	void inject(Frag1ScreenFragment fragment);
	void inject(Frag2ScreenFragment fragment);
	void inject(Frag3ScreenFragment fragment);
	void inject(Frag4ScreenFragment fragment);
	void inject(Frag5ScreenFragment fragment);

	void inject(DrawerScreenFragment fragment);

	@Module
	class FragScreenModule {
	}
}

File: kotlinapp/src/main/java/com/github/mrvilkaman/ui/screens/testfrags/FragScreenComponent.kt
package com.github.mrvilkaman.ui.screens.testfrags

import com.github.mrvilkaman.di.PerScreen
import dagger.Module

@PerScreen
//@Component(dependencies = arrayOf(ActivityComponent::class),
//        modules = arrayOf(FragScreenComponent.FragScreenModule::class))
interface FragScreenComponent {
    fun inject(fragment: Frag1ScreenFragment)
    fun inject(fragment: Frag2ScreenFragment)
    fun inject(fragment: Frag3ScreenFragment)
    fun inject(fragment: Frag4ScreenFragment)
    fun inject(fragment: Frag5ScreenFragment)

    @Module
    class FragScreenModule
}